### PR TITLE
AK: Add operator""_{short_,}string to create a String from a literal

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/DevTools/Profiler/main.cpp [hunk 1]
```diff
@@ -338,7 +338,7 @@ static bool prompt_to_stop_profiling(pid_t pid, DeprecatedString const& process_
 bool generate_profile(pid_t& pid)
 {
     if (!pid) {
-        auto process_chooser = GUI::ProcessChooser::construct("Profiler"sv, String::from_utf8_short_string("Profile"sv), Gfx::Bitmap::load_from_file("/res/icons/16x16/app-profiler.png"sv).release_value_but_fixme_should_propagate_errors());
+        auto process_chooser = GUI::ProcessChooser::construct("Profiler"sv, "Profile"sv), Gfx::Bitmap::load_from_file("/res/icons/16x16/app-profiler.png"_short_string.release_value_but_fixme_should_propagate_errors());
```
Comment (awesomekling) [inline Userland/DevTools/Profiler/main.cpp:341]: Looks like some kind of accident happened here 😅

File: Tests/AK/TestFlyString.cpp [hunk 2]
```diff
@@ -68,15 +68,15 @@ TEST_CASE(long_string)
 
 TEST_CASE(from_string_view)
 {
-    auto fly1 = MUST(FlyString::from_utf8("thisisdefinitelymorethan7bytes"sv));
+    auto fly1 = MUST(Fly"thisisdefinitelymorethan7bytes"_string);
```
Comment (MacDue) [inline Tests/AK/TestFlyString.cpp:71]: Find and replace error, or weird new syntax?

**Review by alimpfard — COMMENTED** [review]
me likey

File: Tests/AK/TestFlyString.cpp (hunk 2 above)
Comment (alimpfard) [reply Tests/AK/TestFlyString.cpp:71]: That's a find-replace error.

File: AK/String.h [hunk 3]
```diff
@@ -259,3 +259,21 @@ struct Formatter<String> : Formatter<StringView> {
...
+// See: https://stackoverflow.com/questions/68789984/immediate-function-as-default-function-argument-initializer-in-clang
+#if defined(AK_COMPILER_CLANG)
+#    define AK_SHORT_STRING_LITERAL_CONSTEVAL constexpr
+#else
+#    define AK_SHORT_STRING_LITERAL_CONSTEVAL consteval
+#endif
+
+[[nodiscard]] ALWAYS_INLINE AK_SHORT_STRING_LITERAL_CONSTEVAL AK::String operator""_short_string(char const* cstring, size_t length)
+{
+    return String::from_utf8_short_string(AK::StringView(cstring, length));
```
Comment (AtkinsSJ) [inline AK/String.h:278]: Can we static-assert here that length is short enough to fit?

Comment (alimpfard) [reply AK/String.h]: The `String::from_utf8_short_string()` function itself asserts that.

File: Userland/DevTools/Profiler/main.cpp (hunk 1 above)
Comment (linusg) [reply Userland/DevTools/Profiler/main.cpp]: oops, my regex skills were not prepared for this!

File: AK/String.h (hunk 3 above)
Comment (trflynn89) [reply AK/String.h]: `from_utf8_short_string` raises a compiler error itself if it's too long

Comment (AtkinsSJ) [reply AK/String.h]: Oh cool. :+1:

File: Tests/AK/TestFlyString.cpp (hunk 2 above)
Comment (linusg) [reply Tests/AK/TestFlyString.cpp]: yep, added `_fly_string` for this as well

File: AK/String.h (hunk 3 above)
Comment (linusg) [reply AK/String.h]: We cannot:

```
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h: In function ‘consteval AK::String operator""_short_string(const char*, size_t)’:
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h:278:26: error: non-constant condition for static assertion
  278 |     static_assert(length <= String::MAX_SHORT_STRING_BYTE_COUNT);
      |                   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/linus/Dev/serenity/Meta/Lagom/../../AK/String.h:278:26: error: ‘length’ is not a constant expression
```

Comment (linusg) [reply AK/String.h]: Ah, right, I guess because it's `consteval` the upstream `VERIFY()` does the trick!

**Comment by Hendiadyoin1** [comment]
The suffixes seem a bit wordy,
Can we just use
`""s` and `""ss`, inline with the `""sv` one we have?

**Comment by linusg** [comment]
>  ""s and ""ss, in line with the ""sv one we already have?

No, clarity is much preferred over short identifiers.

**Comment by alimpfard** [comment]
> Can we just use `""s` and `""ss`, in line with the `""sv` one we already have?

While `operator""s` is valid due to its use in std::, `operator""ss` is not valid and will make clang angry at you.
